In [19]:
import cv2

workdir = '/mnt/1T/baos1/GCA/data/MXIF/cyclegan/'
he_dim = 0.5036
workdir_mxif_resize = '/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space/'

In [20]:
sample_id = 'GCA002ACB'
img = cv2.imread('%s/%s_ALL.tif' % (workdir,sample_id))
h,w,dim = img.shape
print(img.shape)
dim = (int(w * 0.32430937910244567/he_dim), int(h * 0.32411575385040403 / he_dim))
  
# resize image
resized = cv2.resize(img, dim, interpolation = cv2.INTER_AREA)
print(resized.shape)
print('%s/%s_ALL_IN_IHE.tif' % (workdir_mxif_resize,sample_id))
cv2.imwrite('%s/%s_ALL_IN_IHE.tif' % (workdir_mxif_resize,sample_id),resized)

(14479, 20567, 3)
(9318, 13244, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA002ACB_ALL_IN_IHE.tif


True

In [21]:
def resize_mxif_to_ihe_space(x_size,y_size,sample_id):
    print(sample_id)
#     sample_id = 'GCA002ACB'
    img = cv2.imread('%s/%s_ALL.tif' % (workdir,sample_id))
    h,w,dim = img.shape
    print(img.shape)
    dim = (int(w * x_size/he_dim), int(h * y_size / he_dim))

    # resize image
    resized = cv2.resize(img, dim, interpolation = cv2.INTER_AREA)
    print(resized.shape)
    print('%s/%s_ALL_IN_IHE.tif' % (workdir_mxif_resize,sample_id))
    cv2.imwrite('%s/%s_ALL_IN_IHE.tif' % (workdir_mxif_resize,sample_id),resized)

In [22]:
# resize MxIF
resize_mxif_to_ihe_space(0.32430937910244567 ,0.32411575385040403,'GCA002ACB')
resize_mxif_to_ihe_space(0.32431821274665457 ,0.32412503668916937,'GCA002TIB')
resize_mxif_to_ihe_space(0.32432108239095314 ,0.32412536691601174,'GCA003ACA')
resize_mxif_to_ihe_space(0.3243145161290323 ,0.32412004992079874,'GCA003TIB')
resize_mxif_to_ihe_space(0.3243017666392769 ,0.32411196307600687,'GCA004TIB')
resize_mxif_to_ihe_space(0.32430597077244255 ,0.3241287986704653,'GCA011ACB')
resize_mxif_to_ihe_space(0.32430687474077147 ,0.3241229633959231,'GCA011TIB')
resize_mxif_to_ihe_space(0.32430941507114 ,0.3241153973509934,'GCA012ACB')
resize_mxif_to_ihe_space(0.32431067048081824 ,0.32411197154821314,'GCA012TIB')

(14479, 20567, 3)
(9318, 13244, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA002ACB_ALL_IN_IHE.tif
(17035, 13227, 3)
(10963, 8518, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA002TIB_ALL_IN_IHE.tif
(10561, 9904, 3)
(6797, 6378, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA003ACA_ALL_IN_IHE.tif
(20833, 18724, 3)
(13408, 12058, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA003TIB_ALL_IN_IHE.tif
(13433, 19472, 3)
(8645, 12539, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA004TIB_ALL_IN_IHE.tif
(16848, 23950, 3)
(10843, 15423, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA011ACB_ALL_IN_IHE.tif
(13687, 19288, 3)
(8809, 12421, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA011TIB_ALL_IN_IHE.tif
(12080, 22772, 3)
(7774, 14664, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/mxif_in_ihe_space//GCA012ACB_ALL_IN_IHE.tif
(17433, 25519, 3)
(11219, 16433, 3)
/mnt/1T/baos1/GCA/data/MXIF/cyclegan

MxIF physical size

0.32430937910244567 0.32411575385040403

0.32431821274665457 0.32412503668916937

0.32432108239095314 0.32412536691601174 

0.3243145161290323 0.32412004992079874

0.3243017666392769 0.32411196307600687

0.32430597077244255 0.3241287986704653

0.32430687474077147 0.3241229633959231

0.32430941507114 0.3241153973509934

0.32431067048081824 0.32411197154821314

In [27]:
# get H&E and resize
#!/usr/bin/env python
# coding: utf-8

# In[1]:


import openslide

import sys

# sys.path.insert(0, "/home/remedilw/code/bf_translate_openslide/src/mypkg")

# older verison of library for debugging
# sys.path.insert(0, "/home/remedilw/code/Bioformats_Rainy_Pipeline/src/mypkg")

from PIL import ImageChops
from copy import deepcopy

import matplotlib
import javabridge
import bioformats
import openslide
import matplotlib.pyplot as plt
import pathlib as pl
import numpy as np
from PIL import Image
from PIL import ImageChops
import xml.dom.minidom
import xml.etree.ElementTree as ET
import random
from tqdm.notebook import tqdm

#get_ipython().run_line_magic('matplotlib', 'inline')


# In[2]:


# Start javabridge VM
def setup_javabridge():
    myloglevel="ERROR"  # user string argument for logLevel.

    javabridge.start_vm(class_path=bioformats.JARS)

    rootLoggerName = javabridge.get_static_field("org/slf4j/Logger","ROOT_LOGGER_NAME", "Ljava/lang/String;")
    rootLogger = javabridge.static_call("org/slf4j/LoggerFactory","getLogger", "(Ljava/lang/String;)Lorg/slf4j/Logger;", rootLoggerName)
    logLevel = javabridge.get_static_field("ch/qos/logback/classic/Level",myloglevel, "Lch/qos/logback/classic/Level;")
    javabridge.call(rootLogger, "setLevel", "(Lch/qos/logback/classic/Level;)V", logLevel)




    
from IPython.core.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))

setup_javabridge()

def shutdown_vm():
    javabridge.kill_vm()


In [28]:
import glob
f_list = glob.glob('%s/*svs' % workdir)

for i in f_list:
    print(i)
    fileName = i.split('/mnt/1T/baos1/GCA/data/MXIF/cyclegan/')[1].split('.svs')[0]
#     new_name = convert_name(sampleId)
# #     print(new_name)
#     if new_name == 'GCA003TIB':
#         print('hehe')
# #     if new_name in keep_list:
    with bioformats.ImageReader(i) as reader:
        roi_1 = reader.read(c = None, z = 0,t = 0,XYWH = None, series = 1-1,rescale = False )
#             print(roi_1.shape)
        h,w,dim = roi_1.shape
        w = w/3
#             print(w)
        roi_1_crop = roi_1[0:h,0:int(w),:]
        print('/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/%s_cropped.tiff' % fileName)
        cv2.imwrite('/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/%s_cropped.tiff' % fileName ,roi_1_crop)

/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA002ACB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA002ACB_IHE_cropped.tiff
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA002TIB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA002TIB_IHE_cropped.tiff
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA003ACB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA003ACB_IHE_cropped.tiff
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA003TIB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA003TIB_IHE_cropped.tiff
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA004TIB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA004TIB_IHE_cropped.tiff
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA011ACB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA011ACB_IHE_cropped.tiff
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/GCA011TIB_IHE.svs
/mnt/1T/baos1/GCA/data/MXIF/cyclegan/cropped_ihe_one_third/GCA011TIB_IHE_cropped.tiff
/mnt/1T/baos1

### HE rotate and selective roi information
rotate,x,y,w,h
GCA002ACB: rotate right 180, 11392,2064,7520,5312

GCA002TIB: rotate right 180, 11200,5920,6432,4848

GCA003ACA: rotate right 270, 6276, 14248, 788, 1092

GCA003TIB: rotate right 270, 528, 368, 9696, 11920

GCA004TIB: rotate right 180, 5968, 1552, 10944, 7664

GCA011ACB: rotate right 270, 3008, 10460, 15296, 10240

GCA011TIB: rotate right 270, 8568, 13584, 5016, 6528

GCA012ACB

GCA012TIB

### MxIF selective roi information
x,y,w,h
GCA002ACB: 5028,3480,7956,5292

GCA002TIB: 256,6352.6160,4368

GCA003ACA: 4250,3141,1016,1152

GCA003TIB: 96,640, 8320, 12144

GCA004TIB: 0,0, 10956, 8645

GCA011ACB: no need

GCA011TIB: 7584, 1812, 4224, 6072

GCA012ACB

GCA012TIB